client/my-sites/controller.js

```diff
@@ -112,46 +111,62 @@ function renderNoVisibleSites( context ) {
 	);
 }
 
-function isPathAllowedForDomainOnlySite( pathname, domainName ) {
+function renderSelectedSiteIsDomainOnly( { reactContext, selectedSite } ) {
+	const EmptyContentComponent = require( 'components/empty-content' );
+	const { store: reduxStore } = reactContext;
+
+	removeSidebar( reactContext );
+
+	renderWithReduxStore(
+		React.createElement( EmptyContentComponent, {
+			title: i18n.translate( 'This feature is not available for domains' ),
+			line: i18n.translate( 'To use this feature you need to create a site' ),
+			action: i18n.translate( 'Create New Site' ),
+			actionURL: '//dashboard.wordpress.com/wp-admin/index.php?page=my-blogs',
+			secondaryAction: i18n.translate( 'Domain Management' ),
+			secondaryActionURL: domainManagementList( selectedSite.slug )
+		} ),
+		document.getElementById( 'primary' ),
+		reduxStore
+	);
+}
+
+function isPathAllowedForDomainOnlySite( { reactContext: { pathname }, selectedSite } ) {
```

Comment:
I already mention it on Slack but could we avoid destructuring in function signatures? This makes the code more difficult to understand - here I have no idea what parameter this function expects.

```
function isPathAllowedForDomainOnlySite( someObject ) {
  const { reactContext: { pathname }, selectedSite } = someObject;

  ...
```